client/app/scripts/superdesk/editor/spellcheck/spellcheck.js

```diff
@@ -98,36 +85,24 @@ function SpellcheckService($q, api, dictionaries, editor) {
         return elem.classList.contains(ERROR_CLASS);
     };
 
+    function fetchDict() {
+        if (!lang) {
+            dict = $q.reject();
+            return;
+        }
+
+        dict = dictionaries.getActive(lang).then(function(items) {
+            dict = {};
+            angular.forEach(items, addDict);
+            return dict;
+        });
+    }
```

Comment:
`dict` is initialized to an empty object when the service is instantiated, but after this is called it becomes a promise. Should we perhaps initialize it to a promise object (or at least `null`) to avoid confusion about its type when glancing over the code?